snakemake/wrangler_by_sample_finish.smk

```diff
@@ -1,81 +1,159 @@
-		in_scripts='scripts'
+	params:
+		output_dir='/opt/analysis/analysis',
+#		wrangler_dir=output,
+#		fastq_dir=config['fastq_dir']
+	resources:
+		time_min=240
 	output:
-		setup_snakefile=output+'/snakemake_params/setup_run.smk',
-		finish_snakefile=output+'/snakemake_params/finish_run.smk',
-		output_configfile=output+'/snakemake_params/wrangler_by_sample.yaml',
-		out_scripts=directory(output+'/snakemake_params/scripts')
+		output+'/analysis/{sample}/{sample}_mipExtraction/log.txt'
 	shell:
 		'''
-		cp {input.setup_snakefile} {output.setup_snakefile}
-		cp {input.finish_snakefile} {output.finish_snakefile}
-		cp {input.input_configfile} {output.output_configfile}
-		cp -r {input.in_scripts} {output.out_scripts}
+		MIPWrangler mipExtractByArm --masterDir {params.output_dir} --sample {wildcards.sample} --overWriteDirs --minCaptureLength=30
 		'''
+if config['downsample_umi_count']<2**32:
+	rule mip_barcode_correction:
+		input:
+			good_samples=expand(output+'/analysis/{sample}/{sample}_mipExtraction/log.txt', sample=all_samples)
+		params:
+			output_dir='/opt/analysis/analysis',
+#			wrangler_dir=output,
+#			sif_file=config['miptools_sif'],
+			downsample_seed=config['downsample_seed'],
+			downsample_amount=config['downsample_umi_count']
+		resources:
+			mem_mb=config['memory_mb_per_step'],
+			time_min=20
+		output:
+			barcode_corrections_finished=output+'/analysis/{sample}/{sample}_mipBarcodeCorrection/barcodeFilterStats.tab.txt'
+		shell:
+			'''
+			MIPWrangler mipBarcodeCorrection --masterDir {params.output_dir} \
+			--downSampleAmount {params.downsample_amount} --downSampleSeed \
+			{params.downsample_seed} --overWriteDirs --sample {wildcards.sample}
+			'''
+else:
+	rule mip_barcode_correction:
+		input:
+			good_samples=expand(output+'/analysis/{sample}/{sample}_mipExtraction/log.txt', sample=all_samples)
+		params:
+			output_dir='/opt/analysis/analysis',
+#			wrangler_dir=output,
+#			sif_file=config['miptools_sif'],
+			downsample_seed=config['downsample_seed'],
+		resources:
+			mem_mb=config['memory_mb_per_step'],
+			time_min=20
+		output:
+			barcode_corrections_finished=output+'/analysis/{sample}/{sample}_mipBarcodeCorrection/barcodeFilterStats.tab.txt'
+		shell:
+			'''
+			MIPWrangler mipBarcodeCorrection --masterDir {params.output_dir} \
+			--doNotDownSample --downSampleSeed \
+			{params.downsample_seed} --overWriteDirs --sample {wildcards.sample}
+			'''
 
-rule generate_mip_files:
-	'''
-	given that I'm repackaging miptools wrangler (so wrangler.sh is not needed)
-	and that the existing generate_wrangler_scripts.py seems unnecessarily
-	convoluted and that only two files are needed by subsequent steps
-	(mipArms.txt and allMipsSamplesNames.tab.txt) I wrote my own
-	script for this. Input is an arms file and a sample sheet. Output is an arms
-	file with rearranged columns and a two column file with names of all mips
-	and names of all samples (with no pairing between columns of any given row).
-	'''
```

Comment:
Ah yes - this rule probably isn't needed if it got moved to the setup_run.smk script